### PR TITLE
feat: Region Selector [WIP]

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

This Pull Request is to introduce the region selector script used to rotate regional deployment as per the discussions within the core team. @AlexanderSehr FYI

### CODE REVIEW

File: avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1 [hunk 1]
```diff
@@ -5,96 +5,111 @@ Set the deployment region for the module deployment
...
 
-.PARAMETER excludedRegions
-Optional. The regions to exclude from the recommended regions list.
-
 .PARAMETER usePairedRegionsOnly
 Optional. If set, only paired regions will be returned.
 
-.PARAMETER useKnownRegionsOnly
+.PARAMETER moduleRoot
 Optional. If set, only known regions will be returned.
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1:12]: The description seems off

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/regions.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (AlexanderSehr) [inline avm/regions.json:1]: I wonder if this PR should also undo the tempLocations introduced for some module tests or if we should do that in a dedicated PR

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/regions.json (hunk 2 above)
Comment (eriqua) [reply avm/regions.json:1]: I'd suggest to include the change in this PR

Comment (eriqua) [reply avm/regions.json:1]: Anyway, as per discussion, this json file should not be needed. We should retrieve the available locations for each resource via PS cmdlet. What we can keep is the logic to exclude specific locations known to have capacity issues. Please correct me if I'm missing something

File: avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1:1]: The script name should probably be updated. It is not anymore the deployment location that we are going to retrieve, but the resource location. The deployment location should be deterministic for all deployments.

Comment (ChrisSidebotham) [reply avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1:1]: This PR is just in draft for me to align with Alex's function but good spot

File: avm/regions.json (hunk 2 above)
Comment (ChrisSidebotham) [reply avm/regions.json:1]: I thought the agreed approach is still to use the json. especially for where we need to do SKU level exceptions for certain regions then we have a centrally controlled file for the module owner.

File: avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1 (hunk 1 above)
Comment (ChrisSidebotham) [reply avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzDeploymentRegion.ps1:12]: WIP

File: avm/regions.json (hunk 2 above)
Comment (eriqua) [reply avm/regions.json:1]: Not sure I got the same, let's sync offline on that

**Comment by ChrisSidebotham** [comment]
Pending Merge #838

File: avm/regions.json (hunk 2 above)
Comment (AlexanderSehr) [reply avm/regions.json:1]: I agree with @eriqua - BUT - think we agreed to keep the script to be able to still outright exclude locations. Together with the other location PR I think we should have a call sometime to finalize this.

Comment (ChrisSidebotham) [reply avm/regions.json:1]: Lets get this booked in

**Comment by ChrisSidebotham** [comment]
closing in favour of #844
